src/server/redirect_service.rs

```diff
@@ -0,0 +1,52 @@
+use futures::future::{self, FutureResult};
+use hyper;
+use hyper::{StatusCode, Uri};
+use hyper::header::{Host, Location};
+use hyper::server::{Request, Response, Service};
+
+pub struct RedirectService {
+    https_port: u16,
+}
+
+impl RedirectService {
+    pub fn new(https_port: u16) -> RedirectService {
+        RedirectService { https_port: https_port }
+    }
+
+    fn rewrite_uri(&self, uri: Uri, host_header: Option<&Host>) -> String {
```

Comment:
Might be good to add a basic test for this method?

Reply:
Yeah, good idea. It was getting late and I was getting tired. :)